site/docs/view-spec.md

```diff
@@ -0,0 +1,256 @@
+| Required | object-type    | Type of object this metadata file is for: "table" or "view". It must be set to "view" for all objects covered in this spec. |
+| Required | location | Location of the view metadata files |
+| Required | current-version-id | Current version of the view. Set to ‘1’ when the view is first created. |
+| Optional | properties | A string to string map of view properties. Contains pre-set properties such as ‘comment’ describing the view, does not contain arbitrary metadata. |
+| Required | versions | An array of structs describing the last few versions of the view. Controlled by the table property: “version.history.num_entries”. See more below. |
+| Required | version-log | An array of structs describing the log of created versions. See more below. |
+| Optional | schemas | A list of schemas, the same as the ‘schemas’ field from Iceberg table spec. |
+| Optional | current-schema-id | ID of the current schema of the view |
+
+
+‘Versions’ is an array of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | version-id | Monotonically increasing id indicating the version of the view. Starts with “1”. |
+| Required | timestamp-ms | Timestamp expressed in ms since epoch at which the version of the view was created. |
+| Required | summary | A string to string map of view properties to track version metadata. This field can be used by engines to store any necessary properties. Two currently required properties are described below. |
+| Required | representations | A list of "representations" as described below. |
+
+
+Note that each version is stored in a separate AVRO file. This is to ensure that the metadata file stays readable in the case the view definition is huge. As a future extension, an engine-agnostic intermediate representation or a serialized abstract syntax tree of the SQL definition may also be stored in each version, exacerbating the problem.
+
+“summary” is a string-string map with the following required keys. Engines may store additional key-value pairs in this map.
+
+| Required/Optional | Key | Value |
+|-------------------|-----|-------|
+| Required | operation |A string value indicating the view operation that caused this metadata to be created. Allowed values are “CREATE” and “REPLACE” |
+| Optional | engine-version | A string value indicating the version of the engine that performed the operation (create / replace) |
+
+“representations” is a list of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | type | A string indicating the type of representation. The only valid choice is "sql". The rest of the fields are interpreted by the type of representation. |
+
+Here are the fields for "sql" representation type:
```

Comment:
It isn't clear from the two tables. Are these all in the same "representation" struct? If so, then it's not really a struct. It's more of an object because the fields can vary.

Reply:
Combined into 1 table.